File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema04Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.junit.jupiter.api.Test

class Schema04Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val engine = Dynabuffers.parse(Schema04Test::class.java.getResourceAsStream("/schema04.dbs"))
        val output = mapOf("m1" to mapOf("s" to "test", ":type" to 1), "m2" to mapOf("t" to "hello world!", ":type" to 0))

        assertMap(engine, mapOf("m1" to mapOf("s" to "test"), "m2" to mapOf("t" to "hello world!")), output)
    }

}

File: dynabuffers-python/tests/usecase/Schema04Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema04Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def test_parse(self):
        engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema04.dbs"))
        map = engine.deserialize(engine.serialize({"m1": {"s":"test"}, "m2" : {"t":"hello world!"}}))

        self.assertEqual(map, {"m1": {"s":"test", ":type":1}, "m2" : {"t":"hello world!", ":type":0}})


if __name__ == "__main__":
    unittest.main()
